zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -1263,6 +1270,48 @@ angular.module('zeppelinWebApp')
 
   };
 
+  var setMapChart = function(type, data, refresh) {
+    var latArr = [],
+        lngArr = [],
+        newmarkers = {};
+        
+    if (!$scope.chart[type]) {
+      var mapChartModel = function(d) {
+        var key = d[1].replace('-', '_');
+        var obj = {};
+        latArr.push(Math.round(parseFloat(d[2])));
+        lngArr.push(Math.round(parseFloat(d[3])));
+        obj[key] = {
+          lat: parseFloat(d[2]),
+          lng: parseFloat(d[3]),
+          message: d[1],
+          focus: true,
+          draggable: false
+        };
+        return obj;
+      };
+
+      for (var i = 0; i < data.rows.length; i++) {
+        var row = data.rows[i];
+        var rowMarker = mapChartModel(row);
+        newmarkers = $.extend(newmarkers, rowMarker);
+      }
+    } 
+
+    $scope.markers = newmarkers;
+    var bounds = leafletBoundsHelpers.createBoundsFromArray([
+      [Math.max.apply(Math, latArr), Math.max.apply(Math, lngArr)],
+      [Math.min.apply(Math, latArr), Math.min.apply(Math, lngArr)]
+    ]);
+    $scope.bounds = bounds;
+
+    // set map chart height
+    var height = $scope.paragraph.config.graph.height;
+    $('#p'+$scope.paragraph.id+'_mapChart').height(height);
+
+    $scope.center = {};
```

Comment:
Yes, "center", sub-directive is needed for the leaflet main directive. Center is fired when the map is initialized.